dbdimp.c

```diff
@@ -1932,10 +1936,30 @@ MYSQL *mysql_dr_connect(
 
 	    mysql_ssl_set(sock, client_key, client_cert, ca_file,
 			  ca_path, cipher);
-#if MYSQL_VERSION_ID >= SSL_VERIFY_VERSION && MYSQL_VERSION_ID <= SSL_LAST_VERIFY_VERSION
+
+#ifdef SSL_MODE_PREFERRED
+	    if ((svp = hv_fetch(hv, "mysql_ssl_mode", 14, FALSE)) &&
+		*svp)
+	    {
+	          ssl_mode_str = SvPV(*svp, lna);
+	      if (strncmp(ssl_mode_str, "disabled", 8) == 0)
+	        ssl_mode = SSL_MODE_DISABLED;
+	      if (strncmp(ssl_mode_str, "preferred", 9) == 0)
+	        ssl_mode = SSL_MODE_PREFERRED;
+	      if (strncmp(ssl_mode_str, "required", 8) == 0)
+	        ssl_mode = SSL_MODE_REQUIRED;
+	      if (strncmp(ssl_mode_str, "verify_ca", 9) == 0)
+	        ssl_mode = SSL_MODE_VERIFY_CA;
+	      if (strncmp(ssl_mode_str, "verify_identity", 15) == 0)
+	        ssl_mode = SSL_MODE_VERIFY_IDENTITY;
+	    }
+	    mysql_options(sock, MYSQL_OPT_SSL_MODE, &ssl_mode);
+	    if (ssl_mode != SSL_MODE_DISABLED)
+	      client_flag |= CLIENT_SSL;
```

Comment:
This should not happen as when you specify `mysql_ssl=1` SSL should be already enabled.